ui/store/actions.js

```diff
@@ -624,196 +625,36 @@ export function signTypedMsg(msgData) {
-      });
-  };
-}
-
-export function gasLoadingStarted() {
-  return {
-    type: actionConstants.GAS_LOADING_STARTED,
-  };
-}
-
-export function gasLoadingFinished() {
-  return {
-    type: actionConstants.GAS_LOADING_FINISHED,
-  };
-}
-
-export function updateSendTokenBalance({ sendToken, tokenContract, address }) {
-  return (dispatch) => {
-    const tokenBalancePromise = tokenContract
-      ? tokenContract.balanceOf(address)
-      : Promise.resolve();
-    return tokenBalancePromise
-      .then((usersToken) => {
-        if (usersToken) {
-          const newTokenBalance = calcTokenBalance({ sendToken, usersToken });
-          dispatch(setSendTokenBalance(newTokenBalance));
-        }
-      })
-      .catch((err) => {
-        log.error(err);
-        updateSendErrors({ tokenBalance: 'tokenBalanceError' });
-      });
-  };
-}
-
-export function updateSendErrors(errorObject) {
-  return {
-    type: actionConstants.UPDATE_SEND_ERRORS,
-    value: errorObject,
-  };
-}
-
-export function setSendTokenBalance(tokenBalance) {
-  return {
-    type: actionConstants.UPDATE_SEND_TOKEN_BALANCE,
-    value: tokenBalance,
-  };
-}
-
-export function updateSendHexData(value) {
-  return {
-    type: actionConstants.UPDATE_SEND_HEX_DATA,
-    value,
-  };
-}
-
-export function updateSendTo(to, nickname = '') {
-  return {
-    type: actionConstants.UPDATE_SEND_TO,
-    value: { to, nickname },
-  };
-}
-
-export function updateSendAmount(amount) {
-  return {
-    type: actionConstants.UPDATE_SEND_AMOUNT,
-    value: amount,
-  };
-}
-
 export function updateCustomNonce(value) {
   return {
     type: actionConstants.UPDATE_CUSTOM_NONCE,
     value,
   };
 }
 
-export function setMaxModeTo(bool) {
-  return {
-    type: actionConstants.UPDATE_MAX_MODE,
-    value: bool,
-  };
-}
-
-export function updateSend(newSend) {
-  return {
-    type: actionConstants.UPDATE_SEND,
-    value: newSend,
-  };
-}
-
-export function updateSendToken(token) {
-  return {
-    type: actionConstants.UPDATE_SEND_TOKEN,
-    value: token,
-  };
-}
-
-export function clearSend() {
-  return {
-    type: actionConstants.CLEAR_SEND,
-  };
-}
-
-export function updateSendEnsResolution(ensResolution) {
-  return {
-    type: actionConstants.UPDATE_SEND_ENS_RESOLUTION,
-    payload: ensResolution,
-  };
-}
-
-export function updateSendEnsResolutionError(errorMessage) {
-  return {
-    type: actionConstants.UPDATE_SEND_ENS_RESOLUTION_ERROR,
-    payload: errorMessage,
-  };
-}
-
 export function signTokenTx(tokenAddress, toAddress, amount, txData) {
   return async (dispatch) => {
     dispatch(showLoadingIndication());
 
     try {
       const token = global.eth.contract(abi).at(tokenAddress);
-      const txPromise = token.transfer(toAddress, addHexPrefix(amount), txData);
-      dispatch(showConfTxPage());
+      token.transfer(toAddress, addHexPrefix(amount), txData);
+      await wait(500);
       dispatch(hideLoadingIndication());
```

Comment:
Why remove the loading indicator before showing the confirmation page? I had thought that was intended to give the confirmation page time to finish any on-mount stuff before it was shown. I'm not really sure though.

Reply:
showConfTxPage is pretty confusing. It doesn't actually show the confTxPage it sets the txId in the appState though (but we don't know the txId here so it gets set to nothing)... also txId is only referenced (to my knowledge) in conf-tx which ... only works for signature requests? 

```
        <Route
          exact
          path={`${CONFIRM_TRANSACTION_ROUTE}/:id?${SIGNATURE_REQUEST_PATH}`}
          component={ConfTx}
        />
```

All of this to say this wasn't an intentional change I just didn't understand what it was doing I'm going to restore the previous order.